File: day01.py
left = []
right = []

with open("input01", "r") as filo:
    for line in filo:
        [f,s] = line.split("   ")
        left.append(int(f))
        right.append(int(s))

left.sort()
right.sort()

x = 0

for i in range(len(left)):
    x = x + abs(left[i] - right[i])

print("Part 1:", x)

summy = 0

for num in left:
    try:
        idx = right.index(num)
    except:
        continue

    count = 0
    while (idx < len(right)):
        if right[idx] == num:
            count = count + 1
            idx = idx + 1
        else:
            break
    
    summy = summy + num * count

print("Part 2:", summy)

File: day01.clj
(use 'clojure.string)

;; Part 1
(let [l (map (fn [line] (read-string (str \[ line \])))
             (split-lines (slurp "input01")))]
  (def l1 (map first l))
  (def l2 (map second l)))

(when (< (:minor *clojure-version*) 11)
  (defn abs [x] (if (pos? x) x (- x))))

(println "Part 1:" (reduce + (map (comp abs -) (sort l1) (sort l2))))

;; Part 2
(let [freqs (frequencies l2)]
  (println "Part 2:" (reduce + (map (fn [n] (* n (get freqs n 0))) l1))))
